mindmaps-engine/src/main/java/io/mindmaps/engine/controller/VisualiserController.java

```diff
@@ -105,11 +109,11 @@ private String matchQuery(Request req, Response res) {
             final JSONArray halArray = new JSONArray();
 
             withGraph(graph).parseMatch(req.queryParams(REST.Request.QUERY_FIELD))
-                    .stream()
+                    .parallelStream()
```

Comment:
Ok I will try. But using parallelStream() we are basically using a threadpool that has (number-of-physical-core-1), so it shouldnt be a huge number, you still think it might become harmful? Each thread would ask for entity.relation() or similar, would this command open multiple transactions or just one?